x-pack/platform/plugins/shared/lens/public/trigger_actions/open_lens_config/create_action.tsx

```diff
@@ -43,21 +38,21 @@ export class CreateESQLPanelAction implements Action<EmbeddableApiContext> {
 
   public async isCompatible({ embeddable }: EmbeddableApiContext) {
     if (!apiIsPresentationContainer(embeddable)) return false;
-    const { isCreateActionCompatible } = await getAsyncHelpers();
-
-    return isCreateActionCompatible(this.core);
+    return this.core.uiSettings.get(ENABLE_ESQL);
   }
 
   public async execute({ embeddable }: EmbeddableApiContext) {
     if (!apiIsPresentationContainer(embeddable)) throw new IncompatibleActionError();
-    const { executeCreateAction } = await getAsyncHelpers();
-    const editorFrameService = await this.getEditorFrameService();
-
-    executeCreateAction({
-      deps: this.startDependencies,
-      core: this.core,
-      api: embeddable,
-      editorFrameService,
+    const lensEmbeddable = await embeddable.addNewPanel<object, LensApi>({
+      panelType: 'lens',
+      serializedState: {
```

Comment:
Moving away from the runtime state as requested by the presentation team.
Also it has the benefit to move the loading logic into the `deserializeState` helper